_docs/Qabel-Register.md

```diff
@@ -0,0 +1,113 @@
+---
+title: "Identity directory"
+---
+# Identity directory
+
+## Abstract
+
+Identities can be publicly shared in an identity directory provided by the
+register server.
+
+All data is sent as JSON (encoded in UTF-8). All data types are defined
+[here](../Qabel-Client-Local-Data#data-types).
+
+## Concept & structures
+
+The identity directory can be queried using private data (*fields*),
+e.g. an email address or phone number. The register server then
+returns matching identites to the requester.
+
+Changes to the directory are submitted as *update requests*. An update
+request can associate or de-associate any number of fields with/from
+an identity (the combination of a key-pair, a drop URL and an
+alias). The association of a particular field with an identity is an
+*entry* in the directory. Entries are not unique.
+
+An identity is always represented using this JSON structure:
+
+    {
+        "public_key": "hex of 32-byte public key",
+        "drop_url": "drop protocol URL",
+        "alias": "user-specified alias"
+    }
+
+### Fields
+
+Currently defined fields are:
+
+email
+: E-Mail address
+
+phone
+: Phone number (must be capable of receiving SMS)
+
+## APIs
+
+### Search
+
+* Resource: /api/v0/search/
+* Method: GET
+* Request data: field-value pairs (`field=value[&field=value...]`)
+* Response data: `[identity, ...]`
```

Comment:
Would be nice to wrap the identities in a hash map: `{"identities": [identity, …]}`

Some libs have problem with parsing json that has a list as the top level object.